pkg/datastore/test/caveat.go

```diff
@@ -29,8 +29,12 @@ func WriteReadDeleteCaveatTest(t *testing.T, tester DatastoreTester) {
 
 	skipIfNotCaveatStorer(t, ds)
 
-	// Dupes in same transaction are fail to be written
 	ctx := context.Background()
+	// Don't fail on writing empty caveat list
+	_, err = writeCaveatsArray(ctx, ds, []*core.CaveatDefinition{})
```

Comment:
I don't think we need this extra method, you can call `writeCaveats(ctx, ds)` without arguments. `nil` slice is an empty slice too.

Reply:
Yeah, I was just testing that out. Took some time getting used to go syntax, thanks for the help.

Also, let me add a note in contributors.md that to run integration test people can do something like
`go test -v --failfast --tags 'ci,docker' -count=1 ./...`. It took me a while to figure that out (and github actions for the W)

Reply:
Sounds good! I noted you removed the len check in spanner, is there any reason spanner does not need it?

Reply:
Not quite sure, but the test didn't fail there. Probably `BufferWrite` works differently on empty mutations